tools/notes-2019-04-30.md

```diff
@@ -0,0 +1,140 @@
+        * GB: To give an example, if running Babel into Wordlab, if doing that as a build, I don't want to have to put in a bunch of wrappers, since that adds to my download size.  Having that mode as an option seems to make sense.  Having fine-grained protocol methods.
+        * NR: We already don't transpile ES modules.
+        * DE: If you're making a module that has external dependencies, what do you do?
+        * RP: The situation is that if you don't know the entire graph, if you have an external import, then you don't know if that part of the graph has top-level await.  So you can't collapse everything into a single subgraph and you have to keep everything in separate modules.  In practice, this means you can't use Rollup, since the point of Rollup is to condense everything into a single module.  So in practice, bundlers will ignore this part of the spec.  So I want to make sure we're going into this with our eyes open.
+        * GB: Let's say you have modules A, B, and C.  A imports B, and both are synchronous modules, and B imports C, your external module, which may have top-level await.  The import of B into A is effectively a synchronous subgraph which may or may not depend on the async C external dependency.
+        * RP: That's not possible to do in the future world by spec.
+        * GB: Wouldn't A import B still happen synchronously?
+        * RP: No.
+        * DE: Once you get to a top-level await, anything that loads higher in the graph load with Promise.all.  I think both GB and RP have said, why aren't those synchronous reactions?  The idea is that we already have a way of grouping reactions in the spec, which is promises.  So making a separate system is not a good idea.
+        * GB: I agree with RP, this is a big footgun.  I think sync graphs remain sync.  Any sync internal chain should be sync.
+        * SS: does it mean the per-file transpilers would need to emit runtime code defensively to support TLA anywhere in the graph?
+        * SC: Would it make things simpler to make all imports async, with promises optionally resolving synchronously?
+        * DE: We thought about that with the microtask queue flush, but there were some undesirable outcomes.
+        * GB: So this seems like it is a spec constraint of having a graph iteration algorithm running in a sync mode, that then switched into an async mode, as opposed to a language or user constraint. Surely there is a way to refactor simply to retain any synchronous chains?
+    * [Decorators](https://github.com/tc39/proposal-decorators/)
```

Comment:
Let's omit this section to the end, since they are my notes from before the meeting, rather than what people said.

Reply:
from 128 on you mean?